examples/allennlp/allennlp_simple.py

```diff
@@ -97,29 +104,38 @@ def objective(trial):
     if DEVICE > -1:
         model.to(torch.device("cuda:{}".format(DEVICE)))
 
-    lr = trial.suggest_loguniform("lr", 1e-1, 1e0)
+    lr = trial.suggest_loguniform("lr", 1e-2, 1e-1)
     optimizer = torch.optim.SGD(model.parameters(), lr=lr)
 
-    iterator = allennlp.data.iterators.BasicIterator(batch_size=10,)
-    iterator.index_with(vocab)
+    data_loader = torch.utils.data.DataLoader(
+        train_dataset, batch_size=10, collate_fn=allennlp.data.allennlp_collate
+    )
+    validation_data_loader = torch.utils.data.DataLoader(
+        valid_dataset, batch_size=10, collate_fn=allennlp.data.allennlp_collate
+    )
 
     serialization_dir = os.path.join(MODEL_DIR, "trial_{}".format(trial.number))
-    trainer = allennlp.training.Trainer(
+    trainer = allennlp.training.GradientDescentTrainer(
         model=model,
         optimizer=optimizer,
-        iterator=iterator,
-        train_dataset=train_dataset,
-        validation_dataset=valid_dataset,
-        patience=3,
-        num_epochs=6,
+        data_loader=data_loader,
+        validation_data_loader=validation_data_loader,
+        validation_metric="+accuracy",
+        patience=None,
+        num_epochs=50,
         cuda_device=DEVICE,
         serialization_dir=serialization_dir,
+        epoch_callbacks=[AllenNLPPruningCallback(trial, TARGET_METRIC)],
     )
     metrics = trainer.train()
-    return metrics["best_validation_accuracy"]
+    return metrics["best_" + TARGET_METRIC]
 
 
 if __name__ == "__main__":
+    random.seed(41)
+    torch.manual_seed(41)
+    numpy.random.seed(41)
```

Comment:
Could you tell me what numpy features with randomness used in this example?

Reply:
Good question.
Actually, I'm not sure whether I should set seed for NumPy,
but allennlp does in `allennlp train` command and I followed it.
https://github.com/allenai/allennlp/blob/master/allennlp/common/util.py#L218-L247

Do you think it could be removed?

Reply:
If AllenNLP does, I think it's okay, we don't need to remove.